test/integration/hotrestart_test.sh

```diff
@@ -7,19 +7,40 @@ set -e
 # TODO(htuch): In this test script, we are duplicating work done in test_environment.cc via sed.
 # Instead, we can add a simple C++ binary that links against test_environment.cc and uses the
 # substitution methods provided there.
-# TODO(henna): Parameterize IPv4 and IPv6 testing.
-HOT_RESTART_JSON="${TEST_TMPDIR}"/hot_restart.json
-cat "${TEST_RUNDIR}"/test/config/integration/server.json |
-  sed -e "s#{{ upstream_. }}#0#g" | \
-  sed -e "s#{{ test_rundir }}#$TEST_RUNDIR#" | \
-  sed -e "s#{{ ip_loopback_address }}#127.0.0.1#" | \
-  sed -e "s#{{ dns_lookup_family }}#v4_only#" | \
-  cat > "${HOT_RESTART_JSON}"
-
+JSON_TEST_ARRAY=()
+
+# Parameterize IPv4 and IPv6 testing.
+if [[ "${ENVOY_IP_TEST_VERSIONS}" == "v4only" ]] || [[ "${ENVOY_IP_TEST_VERSIONS}" == "all" ]] \
+  || [[ -z "${ENVOY_IP_TEST_VERSIONS}" ]]; then
```

Comment:
It would make sense to have this as the first expression to shortcircuit the other testing.